engine/src/flutter/lib/ui/platform_dispatcher.dart

```diff
@@ -1508,6 +1508,216 @@ class PlatformDispatcher {
+///
+/// This class is typically used in conjunction with
+/// [AccessibilityFeatures.highContrast]. In particular, on Windows, when a user
+/// enables high-contrast mode, they may also pick specific colors that should
+/// be used by application user interfaces. While it is common for applications
+/// to use custom color themes and design languages, in high-contrast mode it is
+/// recommended that widgets use system-specified colors to make content more
+/// legible for users.
+///
+/// See also:
+///
+///   * https://drafts.csswg.org/css-color/#css-system-colors
+///   * https://developer.mozilla.org/en-US/docs/Web/CSS/system-color
+///   * https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors
+final class SystemColor {
+  /// Creates an instance of a system color.
+  ///
+  /// [name] is the name of the color. System colors provided by static getters
+  /// in this class, such as [accentColor] and [buttonText], use standard names
+  /// defined by the [W3C CSS specification](https://drafts.csswg.org/css-color/#css-system-colors).
+  ///
+  /// [value] is the color value, if this color name is supported, and null if
+  /// it's unsupported.
+  const SystemColor({required this.name, this.value});
+
+  /// Standard system color name, as defined by W3C CSS specification.
+  ///
+  /// System color names in Flutter are case-sensitive. This is so that color
+  /// names can be easily used as [Map] keys. This is in contrast to CSS, where
+  /// system color names are not case-sensitive. That is, specifying
+  /// `background-color: aCcEnTcOlOr` is equivalent to specifying
+  /// `background-color: AccentColor`.
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  final String name;
+
+  /// The color value used for the color named [name], if supported.
+  ///
+  /// If [isSupported] is false, the [value] is null. If [isSupported] is true,
+  /// the [value] is not null.
+  final Color? value;
+
+  /// Returns true if the current platform provides the system color with the
+  /// given [name].
+  ///
+  /// See also:
+  ///
+  ///   * [platformProvidesSystemColors], which returns whether the current
+  ///     platform provides system colors.
+  bool get isSupported => value != null;
+
+  /// Returns true if the current platform provides system colors.
+  ///
+  /// As of the current release, system colors are supported on web only.
+  ///
+  /// See also:
+  ///
+  ///   * [isSupported], which returns whether a specific color is supported.
+  static bool get platformProvidesSystemColors => false;
+
+  static UnsupportedError _systemColorUnsupportedError() {
+    return UnsupportedError('SystemColor not supported on the current platform.');
+  }
+
+  /// Returns system color named "AccentColor".
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  static SystemColor get accentColor => throw _systemColorUnsupportedError();
```

Comment:
This seems slightly dangerous that it throw for non web platform. Should we make it nullable?

Reply:
The idea is that framework-side code would check `platformProvidesSystemColors` before accessing any of the colors. Alternatively, we could always return a `SystemColor`, but `SystemColor.isSupported` would return false.

Reply:
sgtm